rfc/2020/alertmanager-integration/christyjacob4.md

```diff
@@ -0,0 +1,446 @@
+- Contribution Name: `Alert Manager Integration`
+- Implementation Owner: @christyjacob4
+- Start Date: 2020-03-03
+- Target Date: 
+- RFC PR: [linkerd/gsoc#0000](https://github.com/linkerd/gsoc/pull/0000)
+- Linkerd Issue: [linkerd/linkerd2#1726](https://github.com/linkerd/linkerd2/issues/1726)
+- Reviewers: 
+
+# Summary
+
+[summary]: #summary
+
+Linkerd uses Prometheus for collecting metrics from various endpoints. Out of the box, Prometheus is an open source monitoring solution that gathers time series based numerical data. Your services need to expose an endpoint (/metrics) from which Prometheus can scrape metrics. The Prometheus monitoring suite comes with tools that help enhance its capabilities like Grafana and AlertManager. The current implementation of the Control Plane makes use of Prometheus for scraping endpoints and aggregating metrics.
+
+The primary goal of this project is to integrate the Alertmanager into the current Control Plane so that Prometheus can provide out of the box alerts to the preferred channels.
+AlertManager can be used for grouping alerts, silencing alerts, routing and sending alerts to preferred channels like slack, emails, pagerduty etc. 
+This will allow users and teams to keep track of critical events that occur on their deployments with ease. 
+
+# Problem Statement (Step 1)
+
+[problem-statement]: #problem-statement
+
+- By default, the Linkerd Control Plane doesn't have an alerting mechanism. Prometheus gathers all the metrics which can be visualised through Grafana, but in order to send these alerts to channels, we need to integrate Alertmanager with Prometheus as part of the control plane.
+
+* Once this RFC is implemented, users will have the option to install **Alertmanager** in the control plane using a flag in in `linkerd install`
```

Comment:
I don't see a reason to have a separate flag to enable `alert-manager` installation, as we already have a way to toggle and configure add-ons through the `add-on` config flag.

Adding another flag would unnecessarily cause confusion, as users can only toggle but still have to use the `add-on` flag for configuration/overwriting values.

Reply:
Are you referring to a kind of invocation like this?
```sh
linkerd install --config config.yaml | kubectl apply -f -
```
config.yaml
```yaml
alertmanager:
  enabled: true
```

Reply:
Yep but `addon-config` flag

Reply:
Got it! 👍